lib/Runtime/ByteCode/ByteCodeEmitter.cpp

```diff
@@ -4016,6 +4017,13 @@ void ByteCodeGenerator::StartEmitFunction(ParseNode *pnodeFnc)
                 this->EnsureLetConstScopeSlots(pnodeFnc->sxFnc.pnodeBodyScope, funcInfo);
             }
         }
+
+        if (!paramScope->GetCanMergeWithBodyScope() && bodyScope->GetScopeSlotCount() == 0 && !bodyScope->GetHasOwnLocalInClosure())
+        {
+            // When we have split scope the body scope may be wrongly marked as must instantiate even though the capture occurred
+            // in param scope. This check is to make sure if no capture occurs in body scope make in not must instantiate.
+            bodyScope->SetMustInstantiate(false);
```

Comment:
If the eval is in the param scope body scope does not have to be an object. If the eval is in the body then the scope slot count will be updated accordingly and it will be dynamic so in FindScopeForSym we will anyway return from the body scope.

We cannot have this check in line 3807 because we have to wait till all the EnsureScopeSlots are done before checking the scope's scope slot count.